lib/src/facts/openbsd/dmi_resolver.cc

```diff
@@ -13,7 +13,15 @@ namespace facter { namespace facts { namespace openbsd {
         result.bios_vendor = sysctl_lookup(HW_VENDOR);
         result.uuid = sysctl_lookup(HW_UUID);
         result.serial_number = sysctl_lookup(HW_SERIALNO);
-        result.product_name = sysctl_lookup(HW_PRODUCT);
+        // OpenBSD running as virtual machine within
+        // OpenBSD vmm don't return HW_PRODUCT. For that
+        // case use the HW_VENDOR, to please the
+        // virtualization_resolver
+        if (sysctl_lookup(HW_PRODUCT).length() > 0) {
```

Comment:
Per @caseywilliams 's suggestion, can you refactor this to only call `sysctl_lookup` once?